src/Components/Main/Pages/MainPage/AsideAdditionalInfo/CurrencyForm/CurrencyForm.tsx

```diff
@@ -0,0 +1,223 @@
+
+type PropsType = {
+  isFetchingCurrency: boolean;
+  belarusCoin: string;
+  americaCoin: string;
+  europeCoin: string;
+  russiaCoin: string;
+  ukraineCoin: string;
+  polandCoin: string;
+  currencyError: Nullable<ErrorType>;
+  setCurrencyRequest: () => void;
+  getCurrency: (coin: string, name: string) => void;
+  setFieldValue: SetFieldValueType;
+};
+
+export const CurrencyForm: FC<PropsType> = ({
+  isFetchingCurrency,
+  belarusCoin,
+  americaCoin,
+  europeCoin,
+  russiaCoin,
+  ukraineCoin,
+  polandCoin,
+  currencyError,
+  setCurrencyRequest,
+  getCurrency,
+  setFieldValue,
+}): ReactElement => {
+  const [activeField, setActiveField] = useState<string>('');
+
+  useEffect(() => {
+    if (isFetchingCurrency) {
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.BelarusCoin, 'Загрузка...');
+      if (activeField && activeField !== FieldName.AmericaCoin) {
+        setFieldValue(FieldName.AmericaCoin, 'Загрузка...');
+      }
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.EuropeCoin, 'Загрузка...');
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.RussiaCoin, 'Загрузка...');
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.UkraineCoin, 'Загрузка...');
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.PolandCoin, 'Загрузка...');
+    } else if (currencyError) {
+      setFieldValue(FieldName.BelarusCoin, 'Нет данных');
+      if (activeField && activeField !== FieldName.AmericaCoin) {
+        setFieldValue(FieldName.AmericaCoin, '1');
+      }
+      setFieldValue(FieldName.EuropeCoin, 'Нет данных');
+      setFieldValue(FieldName.RussiaCoin, 'Нет данных');
+      setFieldValue(FieldName.UkraineCoin, 'Нет данных');
+      setFieldValue(FieldName.PolandCoin, 'Нет данных');
+    } else {
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.BelarusCoin, belarusCoin);
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.AmericaCoin, americaCoin);
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.EuropeCoin, europeCoin);
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.RussiaCoin, russiaCoin);
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.UkraineCoin, ukraineCoin);
+      setFieldValueOnCondition(setFieldValue, activeField, FieldName.PolandCoin, polandCoin);
+    }
+    // eslint-disable-next-line
+  }, [isFetchingCurrency]);
+
+  const getDebouncedCurrency = useDebouncedCallback((coin, name) => getCurrency(coin, name), 900);
+
+  const onCoinChange = (name: string) => (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setActiveField(name);
+    setFieldValue(name, value.replace(/([^.]*\.[^.]*)\./g, '$1'));
+
+    if (!value || value.lastIndexOf('.') !== value.length - 1) {
+      setCurrencyRequest();
+      getDebouncedCurrency(value.replace(/([^.]*\.[^.]*)\./g, '$1'), name);
```

Comment:
The same regexp is used twice, so makes sense to create common util here

Reply:
Done!